torch/_decomp/__init__.py

```diff
@@ -34,13 +34,17 @@ def decomposition_decorator(f):
             registry = decomposition_table
 
         def add_op_to_table(aten_op):
-            # Converts aten.foo to aten.foo.default
-            # Done so I can be lazy and not write default on all of these ops
-            if not isinstance(aten_op, torch._ops.OpOverload):
-                op_overload = aten_op.default
+            overloads = []
+            if isinstance(aten_op, torch._ops.OpOverload):
+                overloads.append(aten_op)
             else:
-                op_overload = aten_op
-            registry[op_overload] = f
+                assert isinstance(aten_op, torch._ops.OpOverloadPacket)
```

Comment:
This changes the behavior of existing decomposition registrations right? We now need to add `.default` to all of them?

Reply:
Yes, unless you wanted all the overloads to be hit by your decomp.

Reply:
Are you going to do that in this PR?

Reply:
I can, although no tests started failing, so it seems like it was OK to hit all overloads XD

Reply:
I suspect it's because not many operators fulfill both 1. have multiple non-CompositeImplicit operators, 2. have different semantics for each of them lol.